frontend/public/components/secrets/create-secret.tsx

```diff
@@ -153,50 +154,86 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
       : k8sUpdate(SecretModel, newSecret, metadata.namespace, newSecret.metadata.name)
     ).then(secret => {
       this.setState({inProgress: false});
-      history.push(resourceObjPath(secret, referenceFor(secret)));
+      if (renderinModal){
+        this.props.onCancel();
+        this.props.onSave(secret.metadata.name);
+      } else {
+        history.push(resourceObjPath(secret, referenceFor(secret)));
+      }
     }, err => this.setState({error: err.message, inProgress: false}));
   }
   render() {
     const { secretTypeAbstraction } = this.state;
     const { onCancel = history.goBack } = this.props;
     const title = `${this.props.titleVerb} ${secretDisplayType(secretTypeAbstraction)} Secret`;
-    return <div className="co-m-pane__body">
-      <Helmet>
-        <title>{title}</title>
-      </Helmet>
-      <form className="co-m-pane__body-group co-create-secret-form co-m-pane__form" onSubmit={this.save}>
-        <h1 className="co-m-pane__heading">{title}</h1>
-        <p className="co-m-pane__explanation">{this.props.explanation}</p>
-
-        <fieldset disabled={!this.props.isCreate}>
-          <div className="form-group">
-            <label className="control-label co-required" htmlFor="secret-name">Secret Name</label>
-            <div>
-              <input className="form-control"
-                type="text"
-                onChange={this.onNameChanged}
-                value={this.state.secret.metadata.name}
-                aria-describedby="secret-name-help"
-                id="secret-name"
-                required />
-              <p className="help-block" id="secret-name-help">Unique name of the new secret.</p>
+    return (renderinModal) ?
```

Comment:
I do not particularly like having a single component play two roles. There's even duplicate code inside this render function.
I'd like to see you use a separate component. You may need to refactor a bit to make it so that you can have better control over the outcome of the form. As shown above you have another if else condition there.